mne/inverse_sparse/mxne_inverse.py

```diff
@@ -92,6 +92,17 @@ def _prepare_gain(forward, info, noise_cov, pca, depth, loose, weights,
         raise ValueError('Depth parameter must be positive (got %s).'
                          % depth)
 
+    if len(forward['src']) == 1:
```

Comment:
You can't go by the length, you need to look at the type. Discrete source spaces can be constructed with legitimate normals, for example.